lib/Alchemy/Phrasea/Controller/Api/V1Controller.php

```diff
@@ -2115,14 +2115,60 @@ protected function createStory($data, $schemaStory, $schemaRecordStory)
 
         if (isset($data->{'story_records'})) {
             $recordsData = (array) $data->{'story_records'};
-            foreach ($recordsData as $data) {
-                $this->addOrDelStoryRecord($story, $data, $schemaRecordStory, 'ADD');
-            }
+            $this->addOrDelStoryRecordsFromData($story, $recordsData, 'ADD');
         }
 
         return $story;
     }
 
+    private function addOrDelStoryRecordsFromRequest(Request $request, $databox_id, $story_id, $action)
+    {
+        $content = $request->getContent();
+
+        $data = @json_decode($content);
+
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            $this->app->abort(400, 'Json response cannot be decoded or the encoded data is deeper than the recursion limit');
+        }
+
+        if (!isset($data->{'story_records'})) {
+            $this->app->abort(400, 'Missing "story_records" property');
+        }
+
+        $recordsData = $data->{'story_records'};
+
+        if (!is_array($recordsData)) {
+            $recordsData = array($recordsData);
+        }
+
+        $story = new \record_adapter($this->app, $databox_id, $story_id);
+
+        $records = $this->addOrDelStoryRecordsFromData($story, $recordsData, $action);
+        $result = Result::create($request, array('records' => $records));
+
+        $this->getDispatcher()->dispatch(PhraseaEvents::RECORD_EDIT, new RecordEdit($story));
+
+        return $result->createResponse();
+    }
+
+    private function addOrDelStoryRecordsFromData(\record_adapter $story, Array $recordsData, $action)
+    {
+        $records = array();
+        $schema = $this->getJsonSchemaRetriever()
+            ->retrieve('file://'.$this->app['root.path'].'/lib/conf.d/json_schema/story_record.json');
+        $cover_set = false;
+
+        foreach ($recordsData as $data) {
+            $records[] = $this->addOrDelStoryRecord($story, $data, $schema, $action);
+            if($action === 'ADD' && !$cover_set && $data->{'use_as_cover'} === true) {
```

Comment:
should be `isset($data->{'use_as_cover'}) && true == $data->{'use_as_cover'}` in case `use_as_cover` property is not isset